airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -606,3 +627,220 @@ def object_exists(self, uuid: str | UUID, **kwargs) -> bool:
+                User can modify this behaviour by providing `unique_columns` params.
+            2. skip: skip the existing objects.
+            3. error: raise an error if an existing object is found.
+
+
+        :param data: A single pandas DataFrame or a list of dicts to be ingested.
+        :param class_name: Name of the class in Weaviate schema where data is to be ingested.
+        :param existing: Strategy for handling existing data: 'skip', or 'replace'. Default is 'skip'.
+        :param unique_columns: Columns in DataFrame or keys in dict uniquely identifying each document,
+            required for 'upsert' operations.
+        :param uuid_column: Column with pre-generated UUIDs. If not provided, UUIDs will be generated.
+        :param vector_column: Column with embedding vectors for pre-embedded data.
+        :param batch_config_params: Additional parameters for Weaviate batch configuration.
+        :param tenant: The tenant to which the object will be added.
+        :return: list of UUID which failed to create
+        """
+        import pandas as pd
+
+        if existing not in ["skip", "replace", "error"]:
+            raise ValueError("Invalid parameter for 'existing'. Choices are 'skip', 'replace', 'error'.")
+
+        if isinstance(data, list):
+            data = pd.json_normalize(data)
+
+        if isinstance(unique_columns, str):
+            unique_columns = [unique_columns]
+        elif unique_columns is None:
+            unique_columns = sorted(data.columns.to_list())
+
+        self.log.info("Inserting %s objects.", data.shape[0])
+
+        if uuid_column is None or uuid_column not in data.columns:
+            (
+                data,
+                uuid_column,
+            ) = self._generate_uuids(
+                df=data,
+                class_name=class_name,
+                unique_columns=unique_columns,
+                vector_column=vector_column,
+                uuid_column=uuid_column,
+            )
+        # drop duplicate rows, using uuid_column and unique_columns
+        data = data.drop_duplicates(
+            subset=list({*unique_columns, uuid_column} - {vector_column, None}), keep="first"
+        )
+
+        uuids_to_create = set()
+        existing_uuid, non_existing_uuid = self._check_existing_objects(
+            data=data, uuid_column=uuid_column, class_name=class_name, existing=existing
+        )
+        if existing == "error" and len(existing_uuid):
+            self.log.info("Found duplicate UUIDs %s", " ,".join(existing_uuid))
+            raise ValueError(
+                f"Found {len(existing_uuid)} object with duplicate UUIDs. You can either ignore or replace"
+                f" them by passing 'existing=skip' or 'existing=replace' respectively."
+            )
+        elif existing == "replace":
+            uuids_to_create = existing_uuid.union(non_existing_uuid)
+            self._delete_objects(existing_uuid, class_name=class_name)
```

Comment:
I don't think its safe to assume we want to delete objects before ingesting new ones.  

The default behaviour for ingest is to replace.  From [weaviate docs](https://github.com/weaviate/weaviate-python-client/blob/e7f547f6d0450ccd28b7f66f18443891de9aa308/weaviate/batch/crud_batch.py#L524) `If the UUID of one of the objects already exists then the existing object will be replaced by the new object.`. So better to let weaviate handle the replace.

Reply:
@mpgreg I have updated the logic to determine the objects we want to delete. 

#### For `replace` case 
We are deleting only those objects belonging to a doc if we want to replace that specific doc. Here I think we can safely assume that certain objects belonging to a document can be deleted and new objects can be created later on since the whole operation is idempotent. WDYT?

#### For `skip` case
No doc is removed.

#### For `error` case
No doc is removed.

Reply:
We should mention that replace happens at document level not on chunk level.

Reply:
Now everything happens document level, even the `skip` and `error`, to reflect the same I have changed the method name to `create_or_replace_document_objects`.  But I can be more explicit in the docstring. Will update it.